uvicorn/subprocess.py

```diff
@@ -47,12 +39,13 @@ def subprocess_started(config, target, sockets, stdin_fileno):
                be the `Server.run()` method.
     * sockets - A list of sockets to pass to the server. Sockets are bound once
                 by the parent process, and then passed to the child processes.
-    * stdin_fileno - The file number of sys.stdin, so that it can be reattached
-                     to the child process.
     """
     # Re-open stdin.
-    if stdin_fileno is not None:
-        sys.stdin = os.fdopen(stdin_fileno)
+    # This is required for some debugging environments.
+    try:
+        sys.stdin = open("/dev/stdin")
```

Comment:
Does /dev/stdin exist on all platforms?

Reply:
@florimondmanca Good question. I believe that it exists on OSX, but Windows is a different beast altogether, and I think it will require a special case (based on [this SO answer](https://stackoverflow.com/a/62913474); thank you @euri10 for pointing it out in the first place). Unfortunately, I don't have access to a Windows machine to test it. My goal was to get it to work without crashing first, and then work on a cross-platform solution.

Since the file descriptor for `stdin` is `0` on *nix and [apparently also on Windows](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/fileno?view=msvc-160), I can replace `open("/dev/stdin")` with `os.fdopen(0)` - would that be OK?

Reply:
Follow-up: I went ahead and switched to `os.fdopen(0)`. Let me know if you need me to make any further changes.